### PR TITLE
HBASE-24395 Backport HBASE-20589 to branch-1

### PR DESCRIPTION
**Pull request by cuibo01**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 47s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 30s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 16s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m  1s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m 10s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 15s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  2s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m  7s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   2m 43s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 25s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  1s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  9s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m  9s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 32s |  hbase-server: The patch generated 1 new + 142 unchanged - 0 fixed = 143 total (was 142)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 49s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 53s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m  7s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   4m 31s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 42s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  |  99m 41s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 54s |  The patch does not generate ASF License warnings.  |
|  |   | 159m 48s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestMetaAssignmentWithStopMaster |
|   | hadoop.hbase.TestClusterPortAssignment |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2690 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 49369da16469 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2690/out/precommit/personality/provided.sh |
| git revision | branch-1 / edccfe4 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/1/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/1/testReport/ |
| Max. process+thread count | 4347 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by saintstack — COMMENTED** [review]
Is this backport of HBASE-20589  as suggested up in the JIRA @cuibo01 ?

Do we need getHostnameLowerCase? Just lowercase always whenever we compare?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java [hunk 1]
```diff
@@ -1251,9 +1251,7 @@ protected void tryRegionServerReport(long reportStartTime, long reportEndTime)
     ClusterStatusProtos.ServerLoad sl = buildServerLoad(reportStartTime, reportEndTime);
     try {
       RegionServerReportRequest.Builder request = RegionServerReportRequest.newBuilder();
-      ServerName sn = ServerName.parseVersionedServerName(
-        this.serverName.getVersionedBytes());
-      request.setServer(ProtobufUtil.toServerName(sn));
+      request.setServer(ProtobufUtil.toServerName(this.serverName));
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java:1254]: Good

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java [hunk 2]
```diff
@@ -2288,9 +2286,7 @@ public void abort(final String reason, Throwable cause) {
       if (rssStub != null && this.serverName != null) {
         ReportRSFatalErrorRequest.Builder builder =
           ReportRSFatalErrorRequest.newBuilder();
-        ServerName sn =
-          ServerName.parseVersionedServerName(this.serverName.getVersionedBytes());
-        builder.setServer(ProtobufUtil.toServerName(sn));
+        builder.setServer(ProtobufUtil.toServerName(this.serverName));
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java:2289]: Good

**Review by saintstack — COMMENTED** [review]
Patch seems good.

See questions above. Thanks.

**Comment by cuibo01** [comment]
image

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 38s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m  7s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m  1s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m  8s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 17s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  1s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  2s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m  9s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   2m 41s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 20s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 56s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  9s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m  9s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 35s |  hbase-server: The patch generated 1 new + 142 unchanged - 0 fixed = 143 total (was 142)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 45s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 53s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m  7s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   4m 29s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 43s |  hbase-client in the patch passed.  |
| -1 :x: |  unit  | 100m 34s |  hbase-server in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 52s |  The patch does not generate ASF License warnings.  |
|  |   | 153m 56s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.master.TestMetaAssignmentWithStopMaster |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2690 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 8cac640c6617 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2690/out/precommit/personality/provided.sh |
| git revision | branch-1 / edccfe4 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/2/artifact/out/diff-checkstyle-hbase-server.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/2/testReport/ |
| Max. process+thread count | 4413 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by cuibo01** [comment]
HBASE-19200 Make hbase-client only depend on ZKAsyncRegistry and ZNod…
because 2.x and 1.x connection are the different，so testStopActiveMaster failed.

>2.x  ServerName oldMetaServer = conn.locateRegions(TableName.META_TABLE_NAME).get(0).getServerName();

>1.x ServerName oldMetaServer = conn.locateRegion(HRegionInfo.FIRST_META_REGIONINFO
            .getRegionName()).getServerName();

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 22s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 23s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m  9s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m 15s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 34s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 21s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  7s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 12s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  5s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   5m  1s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 10s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  7s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 14s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m 14s |  the patch passed  |
| -1 :x: |  checkstyle  |   1m 45s |  hbase-server: The patch generated 2 new + 142 unchanged - 0 fixed = 144 total (was 142)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 33s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 37s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 13s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   5m 20s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 52s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 105m 16s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 52s |  The patch does not generate ASF License warnings.  |
|  |   | 165m  5s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2690 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2b1cee33a93d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2690/out/precommit/personality/provided.sh |
| git revision | branch-1 / edccfe4 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/3/artifact/out/diff-checkstyle-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/3/testReport/ |
| Max. process+thread count | 4526 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by saintstack** [comment]
@cuibo01 Do you see the questions above? There is a checkstyle fail you might want to fix. Thank you.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  17m 53s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 2 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   9m 32s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   1m 20s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m 28s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   2m 42s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 11s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 13s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   2m 58s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   4m 43s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  6s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   1m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 13s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   1m 13s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   2m 25s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m  2s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 56s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 14s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   4m 49s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 40s |  hbase-client in the patch passed.  |
| +1 :green_heart: |  unit  | 104m 11s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 40s |  The patch does not generate ASF License warnings.  |
|  |   | 180m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2690 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 7ffce408a331 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-agent/workspace/Base-PreCommit-GitHub-PR_PR-2690/out/precommit/personality/provided.sh |
| git revision | branch-1 / 4640b22 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/4/testReport/ |
| Max. process+thread count | 3671 (vs. ulimit of 10000) |
| modules | C: hbase-client hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2690/4/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by pankaj72981 — APPROVED** [review]
LGTM, @saintstack Sir.. Is there anything else to be done here?

**Comment by cuibo01** [comment]
@saintstack @pankaj72981  pls review，and merge. thx

**Pull request merged**